waspc/package.yaml

```diff
@@ -51,7 +51,6 @@ library:
     - split
     - unordered-containers
     - path
-    - regex-compat
```

Comment:
This package has a bug in the latest version and requires us to add extra deps in stack.yaml and set flags until they fix it. I felt that maintainer is not very involved, so I decided instead to kick it out and use regex-tdfa which we are using already anyway and is more popular. It resulted with slightly more code because regex-tdfa does not have this replacement that regex-compat has, but it is not a big deal.